core/src/parse.rs

```diff
@@ -257,11 +282,11 @@ fn parse_gdlk(input: &str) -> IResult<&str, Vec<Instr>, VerboseError<&str>> {
     // parses the whole program followed by 0 or more whitespace chars
 
     // consume starting whitespace
-    let (input, _) = multispace0(input)?;
+    let (input, _) = comment_or_spaces(input)?;
     // make sure something is there but don't consume the input
     // TODO: make this error message nicer
     peek(alpha1)(input)?;
-    let (input, res) = all_consuming(many0(try_each))(input)?;
+    let (input, res) = all_consuming(many0(try_each_instr))(input)?;
```

Comment:
I just noticed this function doesn't call `parse_body`. It prob should since a body is just a vec of instructions so the program as a whole is a body. Would that be possible/easy?

Reply:
It could work, but i have logic in parse_body that looks for `{` and `}`. I feel like doing it this way makes the most sense imo

Reply:
ah gotcha, that's fine then